### ISSUE TITLE
Expansion causes an elisp error

### ISSUE THREAD

**Issue by lelit** [joaotavora/yasnippet#747]
I have the following snippet, for `rst-mode`, that upon expansion raises a `Symbol's value as variable is void: $4`:
```
# -*- mode: snippet -*-
# name: Create new table field
# key: new-field
# expand-env: ((yas-indent-line 'fixed))
# --
.. _${1:schema}.${2:tablename}.${3:fieldname}:

.. index::
   triple: $3; $1.$2; ${4:domain}

**$3**: :term:\`$4\`
  $0
```

I use it to document a single column in a database table, something like:
```
.. index::
   triple: shelf; somedomain.some_table; boolean_t

**document**: :term:`boolean_t`
  Whether the type is a *document*.
```

but its expansion immediately gives (note also the indentation glitch, maybe related to #739):
```
.. _schema.tablename.fieldname:

.. index::
triple: schema; schema.tablename; domain

   **fieldname**: :term:[yas] elisp error: Symbol’s value as variable is void: $4
```

I tried to investigate, but I was unable to see the peculiarity with the $4 field: it seem I properly escaped the backquotes as stated in the documentation...

**Comment by npostavs** [comment]
Works for me, here's what I got after expansion:
```
.. _schema.tablename.fieldname:

.. index::
triple: fieldname; schema.tablename; domain

**fieldname**: :term:`domain`
  ```
The indentation glitch is indeed another instance of #739, but escaping the backquotes works fine.

Can you reproduce this from [`emacs -Q`][emacs-Q]? Please also post yasnippet, Emacs and OS versions.

[emacs-Q]: http://www.lunaryorn.com/2015/11/29/reproduce-bugs-in-emacs-Q.html

**Comment by lelit** [comment]
Uhm, I'm confused: while it surely bugged me since I wrote it a few weeks ago, until this morning when I decided to recheck it and eventually file this issue, cut&pasting the snippet and the error from my Emacs session... well, I'm not sure what happened but I'm not able to reproduce it anymore! :smile:
Maybe it wasn't (properly?) compiled in .yas-compiled-snippets.el? Dunno...
Anyway, thank you for checking and sorry for the noise!
